larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewLongTracksTool.cc

```diff
@@ -0,0 +1,173 @@
+
+void TwoViewLongTracksTool::FindLongTracks(const MatrixType &overlapMatrix, ProtoParticleVector &protoParticleVector) const
+{
+    ClusterSet usedClusters;
+    ClusterVector sortedKeyClusters;
+    overlapMatrix.GetSortedKeyClusters(sortedKeyClusters);
+
+    for (const Cluster *const pKeyCluster : sortedKeyClusters)
+    {
+        if (!pKeyCluster->IsAvailable())
+            continue;
+
+        unsigned int n0(0), n1(0);
+        MatrixType::ElementList elementList;
+        overlapMatrix.GetConnectedElements(pKeyCluster, true, elementList, n0, n1);
+
+        IteratorList iteratorList;
+        this->SelectLongElements(elementList, usedClusters, iteratorList);
+
+        // Check that elements are not directly connected and are significantly longer than any other directly connected elements
+        for (IteratorList::const_iterator iIter = iteratorList.begin(), iIterEnd = iteratorList.end(); iIter != iIterEnd; ++iIter)
+        {
+            if (TwoViewLongTracksTool::HasLongDirectConnections(iIter, iteratorList))
+                continue;
+
+            if (!TwoViewLongTracksTool::IsLongerThanDirectConnections(iIter, elementList, m_minMatchedSamplingPointRatio, usedClusters))
+                continue;
+
+            ProtoParticle protoParticle;
+            protoParticle.m_clusterList.push_back((*iIter)->GetCluster1());
+            protoParticle.m_clusterList.push_back((*iIter)->GetCluster2());
+            protoParticleVector.push_back(protoParticle);
+
+            usedClusters.insert((*iIter)->GetCluster1());
+            usedClusters.insert((*iIter)->GetCluster2());
+        }
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewLongTracksTool::SelectLongElements(const MatrixType::ElementList &elementList, const pandora::ClusterSet &usedClusters,
+    IteratorList &iteratorList) const
+{
+    for (MatrixType::ElementList::const_iterator eIter = elementList.begin(); eIter != elementList.end(); ++eIter)
+    {
+        if (usedClusters.count(eIter->GetCluster1()) || usedClusters.count(eIter->GetCluster2()))
+            continue;
+
+        if (eIter->GetOverlapResult().GetLocallyMatchedFraction() < m_minMatchedFraction)
+            continue;
+
+        if (eIter->GetOverlapResult().GetMatchingScore() < m_minMatchingScore)
+            continue;
+
+        if (eIter->GetOverlapResult().GetNMatchedReUpsampledSamplingPoints() < m_minMatchedSamplingPoints)
+            continue;
+
+        const TwoViewXOverlap &xOverlap(eIter->GetOverlapResult().GetTwoViewXOverlap());
+
+        if ((xOverlap.GetXSpan0() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction0() > m_minXOverlapFraction) &&
+            (xOverlap.GetXSpan1() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction1() > m_minXOverlapFraction))
```

Comment:
I think the checks on `xOverlap.GetXSpan0()` and `xOverlap.GetXSpan1()` are redundant now that the fraction functions are used; I think those checks can be dropped from the logic.